lib/mysql.js

```diff
@@ -249,6 +240,30 @@ MySQL.prototype.updateOrCreate = MySQL.prototype.save =
       callback(err, data, meta);
     });
   };
+  
+/**
+ * Replace if the model instance exists with the same id or create a new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [callback] The callback function
+ */
+MySQL.prototype.replaceOrCreate = function (model, data, options, callback) {
+  var self = this;
+  this.modifyOrCreate(model, data, options, function(model, data) { return self.buildReplaceFields(model, data); }, callback);
```

Comment:
Just clarifying: `buildReplaceFields` is a protected method implemented in `loopback-connector`; please see [this PR](https://github.com/strongloop/loopback-connector/pull/35)